Doc/library/urllib.parse.rst

```diff
@@ -543,6 +555,53 @@ operating on :class:`bytes` or :class:`bytearray` objects:
 
    .. versionadded:: 3.2
 
+Special URL Behaviors and Scheme Classes
```

Comment:
The name "class" is confusing to me, because it makes me think of a class statement.

Also, why not just add three boolean parameters to urlparse and urljoin? That should make the behavior simpler to use for users.

Reply:
See also my alternate proposal for a dedicated function on the ticket.
(Fits with the API design guideline of avoiding a boolean param that changes behaviour — I forget the exact phrasing and origin of that)

Reply:
@JelleZijlstra Three boolean parameters would be a pain.  `re.match`, `re.search`, etc., use an `options` parameter to collect all the flags in one.

Reply:
Likewise, I am not sure that a bitfield is the best thing for a regular Python API.  The re module had a backward compat concern that doesn’t apply here.  But (sorry to repeat my opinion) I am not in favour of piecemeal control, I would rather have a universal parsing function that just implements the universal resource identifier spec :)

Reply:
@merwok It's not a bitfield, it's a flag.  ;-)  The three different proposed boolean options all affect the same thing, so a group of choices, a.k.a. an enum, is a good choice -- however, since several choices can be active at once, a flag is a better choice.  Backwards compatibility aside, image if `re.match' looked like:

    def match(need, haystack, ascii=False, multiline=False, ignorecase=False, locale=False, dotall=False, verbose=False, template=False, debug=False):

Reply:
That sounds pretty nice to me! It means I can just look at the signature of `re.match` to know what options I can pass, instead of having to look up a separate set of flags.

It's true though that this sort of API makes it harder to pass around the flags as one unit.

Reply:
Same for me!  I don’t remember having to pass re flags around, and if I needed to I could always build a dict for use as varkwargs.

Reply:
See ticket: 

>  urllib is a module that pre-dates the idea of universal parsing for URIs, where the delimiters (like ://) are enough to determine the parts of a URI and give them meaning (host, port, user, path, etc).
> Backward compat for urllib is always a concern; someone said at the time that it could be good to have a new module for modern, generic parsing, but that hasn’t happened.  Maybe a new parse function, or new parameter to the existing one, could be easier to add.

So I don’t want to have to pass parameters to request standard parsing for this or that component, I only want one function that forgets the special registries that urllib.parse relies on and just parses a well-formed URI into the universal components (scheme host path etc).

Reply:
> So I don’t want to have to pass parameters to request standard parsing for this or that component, I only want one function that forgets the special registries that urllib.parse relies on and just parses a well-formed URI into the universal components (scheme host path etc).

With the currently pushed code, this would be a one line lambda:

```python
from urllib.parse import SchemeFlag, urlparse

parse_standardized = lambda url: urlparse(url, flags=SchemeFlag.NETLOC | SchemeFlag.RELATIVE | SchemeFlag.PARAMS)
```

Is this worth adding to the PR? I imagine it could be done more extremely (i.e., completely redefine the behavior of urlparse), but would backwards compat not be a concern?

Reply:
No, we should not redefine the behaviour of urlparse.

I was always talking about adding another function.  Yes it can be a one-liner, but my point is that I don’t see the usefulness of having the separate flags to pick and choose parts of standard parsing.

Reply:
@merwok - added comments to issue; discussion here should be about this PR, discussion about competing PRs (even if the other PRs haven't been written yet ;) should be on the issue tracker.